lib/scottkit/game.rb

```diff
@@ -283,14 +305,12 @@ def dputs(level, *args) #:nodoc:
     # function is that its behaviour is influenced by the game's
     # options.)
     #
-    def compile_to_stdout(filename, fh = nil)
-      compiler = ScottKit::Game::Compiler.new(self, filename, fh)
-      compiler.compile_to_stdout
+    def compile(out, filename, fh = nil)
```

Comment:
I'm not in love with this argument ordering but it felt like the least bad option considering the optional `fh` argument. If it was me I'd just ditch this method and let the callers instantiate and call the compiler themselves.

Reply:
> If it was me I'd just ditch this method and let the callers instantiate and call the compiler themselves.

I'm not following. What would that look like?

Reply:
I'll respond on the main thread to keep things centrally visible.